src/netaddress.cpp

```diff
@@ -1068,15 +1068,24 @@ CSubNet::CSubNet(const CNetAddr& addr, const CNetAddr& mask) : CSubNet()
 
 CSubNet::CSubNet(const CNetAddr& addr) : CSubNet()
 {
-    valid = addr.IsIPv4() || addr.IsIPv6();
-    if (!valid) {
+    switch (addr.m_net) {
+    case NET_IPV4:
+    case NET_IPV6:
+        valid = true;
+        assert(addr.m_addr.size() <= sizeof(netmask));
+        memset(netmask, 0xFF, addr.m_addr.size());
+        break;
+    case NET_ONION:
+    case NET_I2P:
+    case NET_CJDNS:
+        valid = true;
+        break;
+    case NET_INTERNAL:
+    case NET_UNROUTABLE:
+    case NET_MAX:
         return;
```

Comment:
I think `valid` is left uninitialized here. It's not initialized in `class CSubNet` declaration nor here.

Reply:
It is set to `false` in the default constructor which is called by this constructor:

CSubNet::CSubNet(const CNetAddr& addr) **: CSubNet()**

`netmask` is also set there.

Reply:
Thanks for illuminating. That's kinda sneaky, would prefer to do initialization in the class declaration itself, but no need to do so in this PR.

Reply:
Maybe add `valid = false;` here as belt and suspenders.